File: java/java/src/main/java/org/signal/zkgroup/auth/AuthCredential.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.auth;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class AuthCredential extends ByteArray {

  public static final int SIZE = 181;

  public AuthCredential(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
    Native.AuthCredential_CheckValidContents(contents);
  }

  public byte[] serialize() {
    return contents.clone();
  }

}

File: swift/Sources/SignalClient/zkgroup/AuthCredential.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class AuthCredential: ByteArray {
  public required init(contents: [UInt8]) throws {
    try super.init(contents, checkValid: signal_auth_credential_check_valid_contents)
  }
}
